### PR TITLE
Caching/CAS option improvements

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Update CAS/Caching options and form proper driver options.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

File: include/swift/AST/ModuleDependencies.h [hunk 1]
```diff
@@ -767,6 +768,9 @@ class SwiftDependencyScanningService {
     ModuleDependenciesKindMap ModuleDependenciesMap;
   };
 
+  /// The CASOption created the Scanning Service if used.
+  Optional<clang::CASOptions> CASOpts;
```
Comment (benlangmuir) [inline include/swift/AST/ModuleDependencies.h:772]: Why not `llvm::Optional`?

File: include/swift/Option/Options.td [hunk 2]
```diff
@@ -1812,36 +1812,32 @@ def gcc_toolchain: Separate<["-"], "gcc-toolchain">,
   MetaVarName<"<path>">,
   HelpText<"Specify a directory where the clang importer and clang linker can find headers and libraries">;
 
-def cas_path: Separate<["-"], "cas-path">,
+def compile_cache_job:  Flag<["-"], "compile-cache-job">,
```
Comment (benlangmuir) [inline include/swift/Option/Options.td:1815]: How about `-cache-compile-job` to match clang's `-fcache-compile-job`

File: include/swift/Option/Options.td [hunk 3]
```diff
@@ -1812,36 +1812,32 @@ def gcc_toolchain: Separate<["-"], "gcc-toolchain">,
...
+def compile_cache_remarks:  Flag<["-"], "compile-cache-remarks">,
+  Flags<[FrontendOption, NewDriverOnlyOption]>,
+  HelpText<"Show remarks for compile caching">;
 
-def clang_include_tree: Flag<["-"], "clang-include-tree">,
-  Flags<[FrontendOption, NoDriverOption]>,
-  HelpText<"Use clang include tree">;
+def compile_cache_skip:  Flag<["-"], "compile-cache-skip">,
+  Flags<[FrontendOption, NewDriverOnlyOption]>,
+  HelpText<"Skip loading the compilation result from cache">;
```
Comment (benlangmuir) [inline include/swift/Option/Options.td:1825]: In clang we call this `-fcache-disable-replay`; maybe `-cache-disable-replay`?

File: include/swift/AST/ModuleDependencies.h (hunk 1 above)
Comment (cachemeifyoucan) [reply include/swift/AST/ModuleDependencies.h:772]: This is apparently a mistake. I was intended to update this Optional to llvm::Optional but I accidentally modify the one above.

**Comment by cachemeifyoucan** [comment]
@swift-ci Please clean smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci Please clean smoke test

**Review by artemcm — APPROVED** [review]

**Review by benlangmuir — APPROVED** [review]

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Pull request merged**